json-rpc/src/data.rs

```diff
@@ -55,10 +56,26 @@ pub fn get_metadata(
         if let Some(diem_root) = get_account_state(db, diem_root_address(), version)? {
             metadata_view.with_diem_root(&diem_root)?;
         }
+        metadata_view.diem_version = get_diem_version(db, version)?;
     }
     Ok(metadata_view)
 }
 
+/// Returns the DiemVersion for a specified version, if the version exists.
+pub fn get_diem_version(db: &dyn MoveDbReader, version: u64) -> Result<Option<u64>> {
+    if let Some(config_storage_state) = get_account_state(db, CORE_CODE_ADDRESS, version)? {
+        if let Some(diem_version) = config_storage_state.get_diem_version()? {
+            return Ok(Some(diem_version.major));
+        }
+    }
+    match get_account_state(db, diem_root_address(), version)? {
```

Comment:
Wondering here if it would be better if the configs were stored under the `diem_root_address` as opposed to `0x1` since it wouldn't need to be special-cased then?